Homepage and publications

kidSoba opened the pull request:
Introduces:

- Homepage
- Publications Page
- Neem Overview Pages
- Admin Panel for control over neem-overview and publications content

**IMPORTANT!** Pull-Requests need to be merged in the following order:

1. Minor Style Changes
1. Minor Refactoring
1. Homepage and Publications
1. Temporary News Page

kidSoba committed:
refactor: refactor image name in QA.html

kidSoba committed:
build: add default content-files

kidSoba committed:
feat: add volumes for content files to docker-compose

kidSoba committed:
build: add python dependencies for content-modules

kidSoba committed:
feat: add functions to fetch db-table row counts

kidSoba committed:
refactor: move utility module to helpers-dir

kidSoba committed:
refactor: add helpers.file_handler and refactor some code to use it

kidSoba committed:
refactor: move function to helpers.utility

kidSoba committed:
fix: change number of items per page for neemhub to 8

kidSoba committed:
refactor: remove unused attributes

kidSoba committed:
refactor: change default image path

kidSoba committed:
refactor: move widely used variable to config module

kidSoba committed:
feat: add 'last updated'-field to neems

kidSoba committed:
feat: center pagination links on neemhub-page

kidSoba committed:
fix: adjust footer padding

kidSoba committed:
fix: adjust styling for QA page

kidSoba committed:
docs: update README for local node-module development

kidSoba committed:
feat: add content paths to config

kidSoba committed:
docs: add info about content-pages

kidSoba committed:
feat: add db-model for content settings

kidSoba committed:
feat: add db-model for content data

kidSoba committed:
feat: add download-file function

kidSoba committed:
feat: add thread-handler to helpers

kidSoba committed:
feat: add background-scheduler to helpers

kidSoba committed:
refactor: clean up runserver.py

kidSoba committed:
feat: create downloads-dir on start-up

kidSoba committed:
feat: add neem-overview module

kidSoba committed:
feat: add publications module

kidSoba committed:
feat: add init for content pages on start-up

kidSoba committed:
feat: add content-setting admin panel

kidSoba committed:
feat: add homepage and set it to default page

kidSoba committed:
feat: add link highlighting in nav-bar and publications link

kidSoba committed:
fix: update default neem-overview markdowns to have correct image paths

kidSoba committed:
fix: fix bug in non-debug mode with job-scheduler

jobs were not initializes before being referenced, therefore an error occured

sasjonge commented:
@navidJadid I'm currently checking this PR (but i didn't do the full setup yet with updating files etc). There are currently 2 minor problems I found:

- I would make the Publications Tabs in the top bar the last one, so having Publications after Tutorials
- In Content Management under "Overview Content", if I manually trigger the update there the "Last Update" field is two hours to early. Likely a timezone problem?
- I see on the Newest Part of the Homepage some doubles. I think the reason could be, that some neems could link to the same neemhub page. It would be good to filter that.

kidSoba committed:
refactor: adjust info in footer

kidSoba committed:
fix: move publications-tab in nav-bar

kidSoba committed:
fix: add timezone info to update blocks (hard coded)

kidSoba committed:
docs: adjust comment

kidSoba committed:
fix: change neem-overview update job schedule

change schedule to perform once per day to not block traffic during the rest of the day

kidSoba committed:
refactor: create 'content' folder and move default files there

kidSoba committed:
feat: allow publications bibtex and papers.zip to be accessed from bind mount in addition to via URL

kidSoba commented:
@sasjonge I added all the requested features and made some more changes:

- move publications tab to the end of the navbar
- added timezone information to the 'Last Update' field (it's always UTC+0; everything else didn't work)
- the thing with the doubles on the homepage we decided to let as is after our discussion
- adjusted footer info
- adjusted update schedule for neem-overview files to run only once a day, because otherwise the scheduler would block website access which undesirable to happen several times per day (I tried to have it run in parallel or concurrently, but couldn't manage it not blocking access)
- publications bibtex and papers.zip can now also be provided via bind mound (check the documentation)

I did some testing already, but would kindly ask for others to test it as well to possibly find bugs before publishing. If something comes up, let me know.

sasjonge commented:
I tested this PR and everything looks good now.
